src/main/java/javaslang/collection/Vector.java

```diff
@@ -881,6 +881,15 @@ public int length() {
     }
 
     @Override
+    public Vector<T> padTo(int length, T element) {
+        if(length <= length()) {
+            return this;
+        } else {
+            return appendAll(Stream.gen(() -> element).take(length - length()));
```

Comment:
Now that we have `Iterator` (mostly) up and running, we should use `Iterator` instead of `Stream`. It is more lightweight in the sense of number of objects instantiated. I will create a separate issue for that to add `Iterator.gen(...)` and `Iterator.from(...)`.